src/RulesetEngine.php

```diff
@@ -72,7 +82,16 @@ public function process(
                         continue;
                     }
 
-                    $rules[] = new Violation($dependency, $dependantLayerName, $dependeeLayerName);
+                    $violation = new Violation($dependency, $dependantLayerName, $dependeeLayerName);
+                    $event = new ViolationEvent($violation);
+                    $this->eventDispatcher->dispatch($event);
+
+                    if ($event->isSkipped()) {
```

Comment:
Idea 1: I currently favor this approach, but am worried a bit about the amount of events this would dispatch.

We have an event with a violation (uncovered, allowed, skipped). You can not change the object, but you can tell Deptrac to skip it.

Cons:
 - Listener is quite restricted (but that might be a pro, as it reduces chances of messing things up)
 - We might need to support multiple different actions, e.g. skip, allow, hide, ... which will turn this code here a bit more messy
 - Emitting all these events might have an impact on memory consumption and performance

Pros:
 - Avoids confusing output when the object is altered/replaced
 - Event is very descriptive about what the handler can reasonably do with it